src/main/java/org/opentripplanner/gtfs/mapping/FareProductMapper.java

```diff
@@ -0,0 +1,69 @@
+package org.opentripplanner.gtfs.mapping;
+
+import java.time.Duration;
+import java.util.Currency;
+import org.opentripplanner.ext.fares.model.FareContainer;
+import org.opentripplanner.ext.fares.model.FareProduct;
+import org.opentripplanner.ext.fares.model.RiderCategory;
+import org.opentripplanner.routing.core.Money;
+
+public class FareProductMapper {
+
+  public static int NOT_SET = -999;
+
+  public FareProduct map(org.onebusaway.gtfs.model.FareProduct fareProduct) {
+    var id = AgencyAndIdMapper.mapAgencyAndId(fareProduct.getId());
+    var price = new Money(
+      Currency.getInstance(fareProduct.getCurrency()),
+      (int) (fareProduct.getAmount() * 100)
```

Comment:
Isn't this dependent on the currency. Eg in Yen, the yen is the base unit. See `currency.getDefaultFractionDigits()`

Reply:
I read the documentation and I'm not sure. Does an entry of 100 mean 1 JPY or 100 JPY?

Reply:
Instinctively 100 means 100 yen, not 1 yen. I've implemented this here: https://github.com/leonardehrenfried/OpenTripPlanner/commit/339bb663d07e7bec942043d53ca9a8e8d2a251b3

Do you agree that this is the correct thing to do?

Reply:
100 should be 100 yes, but the currency does not have any cents, so `currency.getDefaultFractionDigits()` returns 0, while for the Euro it returns 2. Otherwise eg. this would return a wrong value https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/main/java/org/opentripplanner/routing/core/Money.java#L33-L38

Reply:
Also this [Javadoc](https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/main/java/org/opentripplanner/routing/core/Money.java#L9-L10) specifies, that the `default number of fraction digits` should be used.

The commit looks correct